core/system_impl.cpp

```diff
@@ -332,6 +334,29 @@ void SystemImpl::add_new_component(uint8_t component_id)
 {
     auto res_pair = _components.insert(component_id);
     if (res_pair.second) {
+        if (component_discovered_callback != nullptr) {
+            ComponentType component_type;
```

Comment:
Small readability thing: I would put that in a function, e.g. `getComponentType(const uint8_t component_id)` (and here do `const auto component_type = getComponentType(component_id)`. What do you think?